avrotest/avrotest.go

```diff
@@ -0,0 +1,59 @@
+package avrotest
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/heetch/avro"
+	"github.com/heetch/avro/avroregistry"
+)
+
+// Register an avro type in a registry for a particular topic and then delete the subject at the end of the test.
+//
+// The following environment variables can be used to
+// configure the connection parameters:
+//
+//	- $KAFKA_REGISTRY_ADDR
+//		The Kafka registry address in host:port
+//		form. If this is empty, localhost:8084 will be used.
+//
+// This requires go1.14 or higher
+func Register(ctx context.Context, t T, x interface{}, topic string) error {
```

Comment:
I'm not entirely sure about this API. It relies on the caller to choose a topic name that's unique, otherwise it might clash with other tests that are being run concurrently. You might want to check out the heetch/kafkatest package which uses a slightly different approach.

Also, I wonder whether "avroregistrytest" might be a better name (given the existing package that's named "avroregistry" and that we're really providing a test utility for the avro registry in particular.

FWIW I had planned to include a similar package, but without the need to rely directly on a real Avro registry instance. It's a simple API that could be implemented with a relatively small amount of code, using an in-memory store, on top of net/http/httptest.
The only wrinkle there is that it would be really nice to have support for compatibility checking, and that isn't yet done in the avro package. See issue #17.

One possibility is to provide a package that does both: depending on the value of an environment variable, it could either talk to a real registry instance, or start a fake server running. I quite like this approach because it means that in the common case you wouldn't have to worry about the requirement to start an Avro registry server instance, but you could still get confidence that your code would work when run against a real instance without changing your tests in any way.

WDYT?

Reply:
:wave: Hey there! :)

> I'm not entirely sure about this API. It relies on the caller to choose a topic name that's unique, otherwise it might clash with other tests that are being run concurrently. You might want to check out the heetch/kafkatest package which uses a slightly different approach.

I used `heetch/kafkatest` to create the unique topic in the tests indeed. Do you want `Register` creates the topic and register it in Schema registry? Could you clarify that bit pretty please? :)

> Also, I wonder whether "avroregistrytest" might be a better name (given the existing package that's named "avroregistry" and that we're really providing a test utility for the avro registry in particular.
>

Yes, I agree it's better package name.

> FWIW I had planned to include a similar package, but without the need to rely directly on a real Avro registry instance. It's a simple API that could be implemented with a relatively small amount of code, using an in-memory store, on top of net/http/httptest.
> The only wrinkle there is that it would be really nice to have support for compatibility checking, and that isn't yet done in the avro package. See issue #17.
> 
> One possibility is to provide a package that does both: depending on the value of an environment variable, it could either talk to a real registry instance, or start a fake server running. I quite like this approach because it means that in the common case you wouldn't have to worry about the requirement to start an Avro registry server instance, but you could still get confidence that your code would work when run against a real instance without changing your tests in any way.
> 
> WDYT?

I love the idea :)